src/content/top-sites.js

```diff
@@ -1,61 +1,26 @@
 "use strict";
 
 (async function () {
+
   function cleanup() {
-    const $content = document.querySelector('.content');
-    const $oldTopSites = $content.querySelector('.topsites');
-    if ($oldTopSites) {
-      $content.removeChild($oldTopSites);
-    }
+    const $topsites = document.querySelector('.top-sites');
+    $topsites.innerHTML = '';
   }
 
   async function loadTopSites() {
-    const $content = document.querySelector('.content');
+    const $topsites = document.querySelector('.top-sites');
+    const $tileTemplate = document.querySelector('#tile-template');
     const topSites = await browser.runtime.sendMessage({
       action: 'getTopSites'
     });
 
-    const $topSitesWrapper = document.createElement('div');
-    $topSitesWrapper.classList.add('topsites');
-    $topSitesWrapper.style.display = 'flex';
-    $topSitesWrapper.style.flexDirection = 'row';
-    $topSitesWrapper.style.margin = '40px 0 0 0';
-    $topSitesWrapper.style.flexWrap = 'wrap';
-    $topSitesWrapper.style.justifyContent = 'center';
-
-    topSites.forEach(site => {
-      const $site = document.createElement('a');
-      $site.setAttribute('href', site.url);
-      $site.style.display = 'flex';
-      $site.style.flexDirection = 'column';
-      $site.style.alignItems = 'center';
-      $site.style.margin = '10px 7px';
-      $site.style.textDecoration = 'none';
-      $site.style.color = 'black';
-
-      const $favicon = document.createElement('img');
-      $favicon.setAttribute('src', site.favicon);
-      $favicon.style.height = '48px';
-      $favicon.style.width = '48px';
-      $favicon.style.boxShadow = 'inset 0 0 0 1px rgba(249, 249, 250, 0.2), 0 1px 8px 0 rgba(12, 12, 13, 0.2)';
-      $favicon.style.transition = 'box-shadow 150ms';
-      $favicon.style.borderRadius = '5px';
-      $favicon.style.backgroundColor = 'white';
-      $site.appendChild($favicon);
-
-      const $title = document.createElement('span');
-      $title.innerText = site.title;
-      $title.style.marginTop = '5px';
-      $title.style.maxWidth = '70px';
-      $title.style.whiteSpace = 'nowrap';
-      $title.style.overflow = 'hidden';
-      $title.style.textOverflow = 'clip';
-      $site.appendChild($title);
-
-      $topSitesWrapper.appendChild($site);
+    topSites.slice(0, 5).forEach(site => {
+      const $tile = $tileTemplate.content.cloneNode(true);
+      $tile.querySelector('a').setAttribute('href', site.url);
+      $tile.querySelector('img').setAttribute('src', site.favicon);
+      $tile.querySelector('span').innerText = site.title;
+      $topsites.appendChild($tile);
```

Comment:
Is this change related?

Reply:
it is - instead of creating speed dials HTML by hand, now we use template from the website, so it is ghosterysearch.com that controls the design.